### PR TITLE
Adds functions to calculate the midpoint_radius representation.

### PR DESCRIPTION
**Pull request by dpsanders**
Adds a version of find_roots that returns a tuple of the midpoints, a tuple of the radii, and a tuple of the symbols

### CODE REVIEW

**Comment by dpsanders** [comment]
Well let's just leave it as `find_roots_midpoint` for the moment and think about a better interface.
This is ready to be merged; the red X comes from coveralls and not from Travis. (I.e., we need more tests...)

**Comment by dpsanders** [comment]
Squashed. Ready to merge I think.

**Comment by lbenet** [comment]
@dpsanders I really like lots of the changes and additions. 

One suggestion: Change the output of `midpoint_radius`. Currently we get:
```julia
julia> @interval 0.1
[0.09999999999999999, 0.1]

julia> midpoint_radius(ans)
(0.1,1.3877787807814457e-17)
```

I think it looks nicer that the output of the last line is something like: 
```
0.1 ± 1.3877787807814457e-17
```

**Comment by lbenet** [comment]
Along the same lines:
```julia
julia> set_interval_precision(BigFloat, 53)
53

julia> @interval 0.1
[9.9999999999999992e-02, 1.0000000000000001e-01]₅₃

julia> midpoint_radius(ans)
(1.0000000000000001e-01,1.3877787807814457e-17)
```
The last line doesn't show that it is a BigFloat interval of 53 bits of precision...

File: src/root_finding/root_finding.jl [hunk 1]
```diff
@@ -1,39 +1,127 @@
 
+# immutable Root{T<:Real}
```
Comment (lbenet) [inline src/root_finding/root_finding.jl]: I really like the idea; have you benchmark it against previous behavior? I would guess it is much faster now, isn't it?

Comment (lbenet) [inline src/root_finding/root_finding.jl]: This was meant on line 14...

**Comment by lbenet** [comment]
I'll wait for your reply about my [comment](https://github.com/dpsanders/ValidatedNumerics.jl/pull/35#issuecomment-113973101). I agree it is ready to be merged.

File: src/root_finding/root_finding.jl (hunk 1 above)
Comment (dpsanders) [inline src/root_finding/root_finding.jl]: I haven't benchmarked it. The change was necessary to have more readable code (and results). I expect it doesn't change the speed too much (at least on v0.4).

**Comment by dpsanders** [comment]
Your two comments about the display are good ones, but not relevant for this PR:

1. The first comment, about output like `3 ± 0.1`, seems to me to be really about how intervals should be displayed. We could add a global option to change this output for displaying any intervals. 

The function `midpoint_radius` is designed only to calculate and return the values for later use; changing its output would require actually making a new type with its own `show`, but really it's just a different representation of the same interval, so this doesn't seem like a good idea. 

I have opened issue #39 for this, but I think we can leave this til after 0.1.

2. The issue about how `BigFloat`s are displayed is due to the recent change https://github.com/JuliaLang/julia/pull/11578 in base Julia; in v0.3 the output is

```julia
(1.0000000000000001e-01 with 53 bits of precision,5.551115123125783e-18 with 53 bits of precision)
```

We can fix this in #39 for intervals; this example with 53 bits suggests that the idea of just removing the indication of the precision in the output of `BigFloat`s was not necessarily a good idea...!

**Comment by dpsanders** [comment]
The red dot is due to the coverage decrease; tests pass on Travis. 
We definitely need better test coverage, but merging for the moment.

**Comment by lbenet** [comment]
I agree with both comments, and that this is merged.

**Pull request merged**